ols/app/models/config.py

```diff
@@ -828,47 +830,81 @@ def __init__(self, **data: Any) -> None:
 class ReferenceContent(BaseModel):
     """Reference content configuration."""
 
+    vector_store_type: Optional[str] = None
     product_docs_index_path: Optional[FilePath] = None
     product_docs_index_id: Optional[str] = None
     embeddings_model_path: Optional[FilePath] = None
+    postgres: Optional[PostgresConfig] = None
 
     def __init__(self, data: Optional[dict] = None) -> None:
         """Initialize configuration and perform basic validation."""
         super().__init__()
         if data is None:
             return
 
+        self.vector_store_type = data.get(
+            "vector_store_type", constants.VectorStoreType.FAISS
+        )
+        valid_vector_store_types = list(constants.VectorStoreType)
+        if self.vector_store_type not in valid_vector_store_types:
```

Comment:
Could it be simplified with a [literal](https://docs.pydantic.dev/latest/api/standard_library_types/#typingliteral) ?

Reply:
The same logic is used for `QueryValidationMethod` ([link](https://github.com/TamiTakamiya/service/blob/TamiTakamiya/pgvector/ols/app/models/config.py#L1050-L1055)). I think we can keep those codes as they are for now and re-visit these issues when we move to python3.12. What do you think?

Reply:
Why add another usage? Pydantic supported literal since 3.9 so we should be safe. Otoh, if the check doesn't work for some reason, I would agree that we can revisit this later.

Reply:
In [constants.py](https://github.com/road-core/service/blob/main/ols/constants.py), we already have [QueryValidationMethod](https://github.com/road-core/service/blob/main/ols/constants.py#L9-L14) and [ModelFamily](https://github.com/road-core/service/blob/main/ols/constants.py#L45-L49), which are defined by extending `StrEnum`. I just thought adding `VectorStoreType` by extending `StrEnum` is more natural than using `typing.Literal`. 

@tisnik What do you think?

Reply:
@TamiTakamiya yup, I used `StrEnum` for this purposes, even before we convert config/models to Pydantic

Reply:
(btw we still need to support Python 3.11, so > 3.12 constructions will be problematic right now)